shared/ui-components/src/forms/FieldGroup.tsx

```diff
@@ -1,57 +1,69 @@
 import React from "react"
 import { ErrorMessage } from "./ErrorMessage"
 
-interface CheckboxSingle {
+interface FieldSingle {
   id: string
   label: string
+  value?: string
   defaultChecked?: boolean
-  register: any
 }
 
-interface CheckboxGroupProps {
+interface FieldGroupProps {
   error?: boolean
-  required?: boolean
   errorMessage?: string
   name: string
-  groupLabel: string
-  fields: CheckboxSingle[]
+  type?: string
+  groupLabel?: string
+  fields: FieldSingle[]
```

Comment:
You could be passing `undefined` to this in a few of the components (when you do a conditional accessor check like `contactPreferenceKeys?.map` - if `contactPrefenceKeys` is `undefined` or `null`, then the variable will become `undefined`, rather than an empty list), I think the type should be `fields?: FieldSingle[]`. (the code works, since you've got a check in the actual render of this component, but I think they types should reflect that - I'm actually surprised Typescript isn't warning you about this/giving an error already...)